ideascube/serveradmin/tests/test_catalog.py

```diff
@@ -1826,3 +1827,65 @@ def test_catalog_list_upgradable_packages(
     assert pkg.version == '2015-09'
     assert pkg.size == '200KB'
     assert isinstance(pkg, ZippedZim)
+
+
+def test_catalog_list_nothandled_packages(
+        tmpdir, settings, testdatadir, mocker):
+    from ideascube.serveradmin.catalog import Catalog
+
+    zippedzim = testdatadir.join('catalog', 'wikipedia.tum-2015-08')
+    path = tmpdir.mkdir('source').join('wikipedia_tum_all_nopic_2015-08.zim')
+    zippedzim.copy(path)
+
+    remote_catalog_file = tmpdir.join('source').join('catalog.yml')
+    with remote_catalog_file.open(mode='w') as f:
+        f.write('all:\n')
```

Comment:
Also, we're pretty much cargo-culting all those tests through copy-pasta at this point, but when I wrote the first few ones, I found `with file.open(...)` was mighty cool and `py.path` was great for offering this API.

Since then, I discovered it also offered a `file.write(...)` API, making all those `with` blocks look silly. :smile: